llvm/lib/Transforms/Utils/FunctionComparator.cpp

```diff
@@ -804,6 +804,10 @@ int FunctionComparator::cmpOperations(const Instruction *L,
         return Res;
     }
   }
+  if (const CastInst *Cast = dyn_cast<CastInst>(L)) {
```

Comment:
Perhaps I'm missing something here, but I'm surprised this isn't already caught by

```
  if (int Res = cmpTypes(L->getType(), R->getType()))
    return Res;
```

above? Wouldn't `L->getType()` be equivalent to `Cast->getDestTy()`?

Reply:
I just asked myself the same question, because it is surprising this requires explicit code, but my understanding is that a return value of `!= 0` means that they different and don't match. So this only returns `Res` here if they are not the same.